scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -315,6 +323,10 @@ class UDPPeerGroup[M](val config: Config)(
 
 object UDPPeerGroup {
 
+  def getMessageQueue[M]() = {
+    ConcurrentQueue.unsafe[Task, M](BufferCapacity.Unbounded(None), monix.execution.ChannelType.SPMC)
```

Comment:
I thought there should be some capacity, to protect ourselves from being DoS'd. Unfortunately that doesn't play very well with the notion of using a channel event to signal the end of processing: if we try to close the channel for any reason, and push an exit event to the queue, if it's full we can't, and we'd either have to:
1. Block by using `queue.push(...)` but we'd be blocking `Channel.close()` which is not what we want.
2. Use `queue.tryPush(...)` but that would fail if the queue is full, so consumers would not be notified, which is bad.
3. Use `queue.push(...).startAndForget` so we can continue closing and also make sure the message eventually gets there and consumers are freed, no resource leaks.

Reply:
Forget what I said here, I think https://github.com/input-output-hk/scalanet/pull/80#issuecomment-677552959 is a better way to signal channel closure and plays nicely with the bounded queues too.